jackson-databind/src/main/java/io/micronaut/jackson/modules/BeanIntrospectionModule.java

```diff
@@ -228,20 +230,55 @@ private static Object suppressableValue(JsonInclude.Value inclusion) {
         );
     }
 
+    @Nullable
+    private PropertyNamingStrategy findNamingStrategy(MapperConfig<?> mapperConfig, BeanIntrospection<?> introspection) {
+        AnnotationValue<JsonNaming> namingAnnotation = introspection.getAnnotation(JsonNaming.class);
+        if (namingAnnotation != null) {
+            Optional<Class<?>> clazz = namingAnnotation.classValue();
+            if (clazz.isPresent()) {
+                try {
+                    Constructor<?> emptyConstructor = clazz.get().getConstructor();
+                    return (PropertyNamingStrategy) emptyConstructor.newInstance();
+                } catch (NoSuchMethodException ignored) {
+                } catch (ReflectiveOperationException e) {
+                    throw new RuntimeException("Failed to construct configured PropertyNamingStrategy", e);
+                }
+            }
+        }
+        return mapperConfig.getPropertyNamingStrategy();
+    }
+
+    private String getName(MapperConfig<?> mapperConfig, @Nullable PropertyNamingStrategy namingStrategy, AnnotatedElement property) {
+        AnnotationValue<JsonProperty> explicitAnnotation = property.getAnnotation(JsonProperty.class);
```

Comment:
Just calling `property.stringValue(JsonProperty.class)` is probably more efficient. You will need to change the argument type to `AnnotationMetadata`